### DISCUSSION TITLE
Should a new schema be created for each request?

### DISCUSSION THREAD

**Discussion by sitepark-veltrup** [graphql-java-kickstart/graphql-java-servlet#464] [Q&A] [answered]
I am using GraphQL servlets for the first time and have a question to understand.

In this [example](https://www.graphql-java-kickstart.com/servlet/getting-started/#create-a-servlet-class) is shown how to build a servlet.

About `getConfiguration()` the schema and `GraphQLConfiguration` is created.

If I understand it correctly you can see [here](https://github.com/graphql-java-kickstart/graphql-java-servlet/blob/master/graphql-java-servlet/src/main/java/graphql/kickstart/servlet/AbstractGraphQLHttpServlet.java#L84) how for each request the schema and the configuration is recreated.

Is this the recommended way? It feels like it is an unnecessary overhead to re-parse the schema for each request.

Thanks in advance

**Comment by oliemansm** [comment, marked as answer]
That's a very basic and simplified example. That's not necessarily the recommended way.

**Reply by sitepark-veltrup** [reply]
Wow, that was a quick response. Thank you very much.
